mailbox/opensearch/src/main/java/org/apache/james/mailbox/opensearch/json/MessageToOpenSearchJson.java

```diff
@@ -90,4 +92,14 @@ public String getUpdatedJsonMessagePart(Flags flags, ModSeq modSeq) throws JsonP
         Preconditions.checkNotNull(flags);
         return mapper.writeValueAsString(new MessageUpdateJson(flags, modSeq));
     }
+
+    public ObjectNode updateMailboxId(ObjectNode origin, MailboxId newMailboxId) {
+        origin.put(JsonMessageConstants.MAILBOX_ID, newMailboxId.serialize());
+        // TODO check JsonMessageConstants.SAVE_DATE, it should be the date of the move or message.getSaveDate() ?
```

Comment:
Should be message.getSaveDate.

To get this info a DB read at METADATA level is acceptable.